controllers/tasks.js

```diff
@@ -156,50 +194,81 @@ const createTask = [
+    let users = usersWithInfo.map(({ provider, ...user }) => {
       return {
         ...user,
         type: provider.startsWith('local-contextual') ? 'light' : 'graasp',
       };
     });
 
-    // fetch app instances, and then append 'settings' key to each app instance object
-    const appInstancesCursor = await fetchAppInstances(
-      itemsCollection,
-      task.spaceId,
-    );
-    const appInstancesArray = await appInstancesCursor.toArray();
-    const appInstances = [];
-    for (let i = 0; i < appInstancesArray.length; i += 1) {
-      // eslint-disable-next-line no-await-in-loop
-      const appInstanceWithSettings = await appendAppInstanceSettings(
-        appInstancesCollection,
-        appInstancesArray[i],
-      );
-      appInstances.push(appInstanceWithSettings);
+    if (task.view === 'compose') {
+      users = users.filter((user) => user.type === 'graasp');
     }
 
-    const appInstanceIds = appInstances.map((appInstance) => appInstance._id);
-    const appInstancesResourcesCursor = fetchAppInstanceResources(
-      appInstanceResourcesCollection,
-      appInstanceIds,
-    );
-
-    // create file name to write data to; create metadata object; write/upload/hide file
+    // create file name to write data to; create metadata object
     const fileName = `${task.createdAt.toISOString()}-${task._id.toString()}.json`;
     const metadata = {
       spaceTree,
       createdAt: new Date(Date.now()),
     };
 
-    writeDataFile(
-      fileName,
-      actionsCursor,
-      users,
-      appInstances,
-      appInstancesResourcesCursor,
-      metadata,
-      () => {
+    // if view not compose, fetch app instances, and append 'settings' key to each app inst. object
+    // then write/upload/hide file w/appInstances and appInstanceResources
+    // else (compose view) write file w/o appInstances and appInstanceResources
+    if (task.view !== 'compose') {
+      const appInstancesCursor = await fetchAppInstances(
+        itemsCollection,
+        task.spaceId,
+      );
+      const appInstancesArray = await appInstancesCursor.toArray();
+      const appInstances = [];
+      for (let i = 0; i < appInstancesArray.length; i += 1) {
+        // eslint-disable-next-line no-await-in-loop
+        const appInstanceWithSettings = await appendAppInstanceSettings(
+          appInstancesCollection,
+          appInstancesArray[i],
+        );
+        appInstances.push(appInstanceWithSettings);
+      }
+
+      const appInstanceIds = appInstances.map((appInstance) => appInstance._id);
+      const appInstancesResourcesCursor = fetchAppInstanceResources(
+        appInstanceResourcesCollection,
+        appInstanceIds,
+      );
+
+      writeLiveViewDataFile(
+        fileName,
+        actionsCursor,
+        users,
+        appInstances,
+        appInstancesResourcesCursor,
+        metadata,
+        () => {
+          uploadFile(
+            `https://graasp.eu/spaces/${task.spaceId}/file-upload`,
```

Comment:
Is this URL not factored out to some config file?

Reply:
We haven't done that so far. Should we? @ahgentil if we put this in a `.env.local` would it affect your ability to release?

Reply:
As a start, it should be placed in an api.js file or something like that. We can then factor it out to a .env file later.

Reply:
In an .env file would be "weird" because the api would have to have env variables for "external" middleware and with no explicit passing of those values. It would make sense if your middleware had "boot" options, and the api would read those from the .env and pass them as options to your middleware. You can explore that.
But maybe, for now, I would stick with a file in your project with those.

Reply:
makes sense! moved all url/path constants into `/config/api.js` and pushed the changes